### ISSUE TITLE
thermald fails to work when running as a systemd service at boot up 

### ISSUE THREAD

**Issue by mschiu77** [intel/thermal_daemon#211]
I'm using thermald as a systemd service at bootup to prevent my laptop from shutdown by overheat.

However, I can still hit this issue and get the journal log message as follows
```
Dec 27 12:07:05 endless kernel: thermal thermal_zone0: critical temperature reached (100 C), shutting down
```

The thermald log during boot shows as follows
```
Jan 16 09:21:43 endless thermald[474]: [WARN]NO RAPL sysfs present
Jan 16 09:21:43 endless thermald[474]: [WARN]27 CPUID levels; family:model:stepping 0x6:7e:5 (6:126:5)
Jan 16 09:21:43 endless thermald[474]: [WARN]Polling mode is enabled: 4
Jan 16 09:21:43 endless thermald[474]: [WARN]Thermal DTS: No coretemp sysfs found
Jan 16 09:21:43 endless thermald[474]: [WARN]sensor id 1 : No temp sysfs for reading raw temp
Jan 16 09:21:43 endless thermald[474]: I/O warning : failed to load external entity "/etc/thermald/thermal-conf.xml"
Jan 16 09:21:43 endless thermald[474]: [WARN]error: could not parse file /etc/thermald/thermal-conf.xml
Jan 16 09:21:43 endless thermald[474]: [WARN]sysfs open failed
Jan 16 09:21:43 endless thermald[474]: I/O warning : failed to load external entity "/etc/thermald/thermal-conf.xml"
Jan 16 09:21:43 endless thermald[474]: [WARN]error: could not parse file /etc/thermald/thermal-conf.xml
Jan 16 09:21:43 endless thermald[474]: [ERR]Thermal DTS or hwmon: No Zones present Need to configure manually
Jan 16 09:21:43 endless thermald[474]: I/O warning : failed to load external entity "/etc/thermald/thermal-conf.xml"
Jan 16 09:21:43 endless thermald[474]: [WARN]error: could not parse file /etc/thermald/thermal-conf.xml
```
I can see the thermald is still running, but it just keeps polling the `fd` until the machine shutdown.

However, if I restart the thermald.service by `systemctl restart thermald.service`, there're no such warning message and I can have the laptop running a continuous CPU intensive task for a whole week w/o problem.

The output message of thermald.service is different
```
[INFO]sensor_update: type x86_pkg_temp
[INFO]sensor_update: type TCPU
[INFO]sensor_update: type INT3400
[INFO]sensor_update: type iwlwifi
[INFO]thd_read_default_thermal_sensors loaded 6 sensors 
[INFO]dts /sys/devices/platform/coretemp.0/name doesn't exist
[WARN]sensor id 9 : No temp sysfs for reading raw temp
[WARN]sensor id 9 : No temp sysfs for reading raw temp
[WARN]sensor id 9 : No temp sysfs for reading raw temp
[DEBUG]ART count 5 ...
[DEBUG]ART length 520 ...
[DEBUG]ART 0: SRC TFN1:	[DEBUG]ART 0: TGT TCPU:	[DEBUG]ART 0: WT 100:
[DEBUG]ART 1: SRC TFN1:	[DEBUG]ART 1: TGT SEN1:	[DEBUG]ART 1: WT 100:
[DEBUG]ART 2: SRC TFN1:	[DEBUG]ART 2: TGT SEN2:	[DEBUG]ART 2: WT 100:
[DEBUG]ART 3: SRC TFN1:	[DEBUG]ART 3: TGT SEN3:	[DEBUG]ART 3: WT 100:
[DEBUG]ART 4: SRC TFN1:	[DEBUG]ART 4: TGT SEN4:	[DEBUG]ART 4: WT 100:
[DEBUG]TRT count 1 ...
[DEBUG]TRT 0: SRC TCPU:	[DEBUG]TRT 0: TGT SEN1:	[DEBUG]TRT 0: INF 10:	[DEBUG]TRT 0: SMPL 100:
[DEBUG]uuid: 97C68AE7-15FA-499c-B8C9-5DA81D606E0A
[DEBUG]uuid: 63BE270F-1C11-48FD-A6F7-3AF253FF3E2D
[DEBUG]uuid: 9E04115A-AE87-4D1C-9500-0F3E340BFE75
[INFO]Passive 1 UUID is not present, hence ignore _TRT, as it may have junk!![INFO]Using config file /etc/thermald/thermal-conf.xml
I/O warning : failed to load external entity "/etc/thermald/thermal-conf.xml"
[WARN]error: could not parse file /etc/thermald/thermal-conf.xml
[INFO]sensor index:2 SEN4 /sys/class/thermal/thermal_zone2/ Async:0 
[INFO]sensor index:0 acpitz /sys/class/thermal/thermal_zone0/ Async:0 
[INFO]sensor index:5 x86_pkg_temp /sys/class/thermal/thermal_zone5/ Async:1 
[INFO]sensor index:3 TCPU /sys/class/thermal/thermal_zone3/ Async:0 
[INFO]sensor index:1 INT3400 /sys/class/thermal/thermal_zone1/ Async:0 
[INFO]sensor index:4 iwlwifi /sys/class/thermal/thermal_zone4/ Async:0 
[INFO]sensor index:6 hwmon /sys/class/hwmon/hwmon4/temp1_input Async:0 
[INFO]sensor index:7 hwmon /sys/class/hwmon/hwmon4/temp2_input Async:0 
[INFO]sensor index:8 hwmon /sys/class/hwmon/hwmon4/temp3_input Async:0
```

I saw similar bug report on https://github.com/intel/thermal_daemon/issues/115 but I get nothing useful. The problem here is that the `/sys/class/powercap` and `/dev/acpi_thermal_rel` are not created yet while thermald starts during boot.

The content of `thermald.service` in the thermald debian package is quite simple
```
[Unit]
Description=Thermal Daemon Service

[Service]
Type=dbus
SuccessExitStatus=1
BusName=org.freedesktop.thermald
ExecStart=/usr/sbin/thermald --no-daemon --dbus-enable

[Install]
WantedBy=multi-user.target
Alias=dbus-org.freedesktop.thermald.service
```
It doesn't require the thermal zone/RAPL information to be ready and that's the reason why it doesn't take any effect for thermal throttling.

If I rebuild the kernel with the following CONFIGs from `m` to `y`, then the thermal throttling work as expected.
```
CONFIG_INT340X_THERMAL=m
CONFIG_ACPI_THERMAL_REL=m

CONFIG_INTEL_RAPL_CORE=m
CONFIG_INTEL_RAPL=m
```

Any suggestion about what I can do except rebuilding the kernel?

Thanks,
Chris

**Comment by spandruvada** [comment]
On Thu, 2020-01-16 at 00:01 -0800, Ming Shuo Chiu wrote:

I'm using thermald as a systemd service at bootup to prevent my laptop from shutdown by overheat.

However, I can still hit this issue and get the journal log message as follows

Dec 27 12:07:05 endless kernel: thermal thermal_zone0: critical temperature reached (100 C), shutting down

The thermald log during boot shows as follows

Jan 16 09:21:43 endless thermald[474]: [WARN]NO RAPL sysfs present

Jan 16 09:21:43 endless thermald[474]: [WARN]27 CPUID levels; family:model:stepping 0x6:7e:5 (6:126:5)

Jan 16 09:21:43 endless thermald[474]: [WARN]Polling mode is enabled: 4

Jan 16 09:21:43 endless thermald[474]: [WARN]Thermal DTS: No coretemp sysfs found

Jan 16 09:21:43 endless thermald[474]: [WARN]sensor id 1 : No temp sysfs for reading raw temp

Jan 16 09:21:43 endless thermald[474]: I/O warning : failed to load external entity "/etc/thermald/thermal-conf.xml"

Jan 16 09:21:43 endless thermald[474]: [WARN]error: could not parse file /etc/thermald/thermal-conf.xml

Jan 16 09:21:43 endless thermald[474]: [WARN]sysfs open failed

Jan 16 09:21:43 endless thermald[474]: I/O warning : failed to load external entity "/etc/thermald/thermal-conf.xml"

Jan 16 09:21:43 endless thermald[474]: [WARN]error: could not parse file /etc/thermald/thermal-conf.xml

Jan 16 09:21:43 endless thermald[474]: [ERR]Thermal DTS or hwmon: No Zones present Need to configure manually

Jan 16 09:21:43 endless thermald[474]: I/O warning : failed to load external entity "/etc/thermald/thermal-conf.xml"

Jan 16 09:21:43 endless thermald[474]: [WARN]error: could not parse file /etc/thermald/thermal-conf.xml

I can see the thermald is still running, but it just keeps polling the fd until the machine shutdown.

However, if I restart the thermald.service by systemctl restart thermald.service, there're no such warning message and I can have the laptop running a continuous CPU intensive task for a whole week w/o problem.

The output message of thermald.service is different

[INFO]sensor_update: type x86_pkg_temp

[INFO]sensor_update: type TCPU

[INFO]sensor_update: type INT3400

[INFO]sensor_update: type iwlwifi

[INFO]thd_read_default_thermal_sensors loaded 6 sensors

[INFO]dts /sys/devices/platform/coretemp.0/name doesn't exist

[WARN]sensor id 9 : No temp sysfs for reading raw temp

[WARN]sensor id 9 : No temp sysfs for reading raw temp

[WARN]sensor id 9 : No temp sysfs for reading raw temp

[DEBUG]ART count 5 ...

[DEBUG]ART length 520 ...

[DEBUG]ART 0: SRC TFN1: [DEBUG]ART 0: TGT TCPU: [DEBUG]ART 0: WT 100:

[DEBUG]ART 1: SRC TFN1: [DEBUG]ART 1: TGT SEN1: [DEBUG]ART 1: WT 100:

[DEBUG]ART 2: SRC TFN1: [DEBUG]ART 2: TGT SEN2: [DEBUG]ART 2: WT 100:

[DEBUG]ART 3: SRC TFN1: [DEBUG]ART 3: TGT SEN3: [DEBUG]ART 3: WT 100:

[DEBUG]ART 4: SRC TFN1: [DEBUG]ART 4: TGT SEN4: [DEBUG]ART 4: WT 100:

[DEBUG]TRT count 1 ...

[DEBUG]TRT 0: SRC TCPU: [DEBUG]TRT 0: TGT SEN1: [DEBUG]TRT 0: INF 10:   [DEBUG]TRT 0: SMPL 100:

[DEBUG]uuid: 97C68AE7-15FA-499c-B8C9-5DA81D606E0A

[DEBUG]uuid: 63BE270F-1C11-48FD-A6F7-3AF253FF3E2D

[DEBUG]uuid: 9E04115A-AE87-4D1C-9500-0F3E340BFE75

[INFO]Passive 1 UUID is not present, hence ignore _TRT, as it may have junk!![INFO]Using config file /etc/thermald/thermal-conf.xml

I/O warning : failed to load external entity "/etc/thermald/thermal-conf.xml"

[WARN]error: could not parse file /etc/thermald/thermal-conf.xml

[INFO]sensor index:2 SEN4 /sys/class/thermal/thermal_zone2/ Async:0

[INFO]sensor index:0 acpitz /sys/class/thermal/thermal_zone0/ Async:0

[INFO]sensor index:5 x86_pkg_temp /sys/class/thermal/thermal_zone5/ Async:1

[INFO]sensor index:3 TCPU /sys/class/thermal/thermal_zone3/ Async:0

[INFO]sensor index:1 INT3400 /sys/class/thermal/thermal_zone1/ Async:0

[INFO]sensor index:4 iwlwifi /sys/class/thermal/thermal_zone4/ Async:0

[INFO]sensor index:6 hwmon /sys/class/hwmon/hwmon4/temp1_input Async:0

[INFO]sensor index:7 hwmon /sys/class/hwmon/hwmon4/temp2_input Async:0

[INFO]sensor index:8 hwmon /sys/class/hwmon/hwmon4/temp3_input Async:0

I saw similar bug report on #115<https://github.com/intel/thermal_daemon/issues/115> but I get nothing useful. The problem here is that the /sys/class/powercap and /dev/acpi_thermal_rel are not created yet while thermald starts during boot.

The content of thermald.service in the thermald debian package is quite simple

[Unit]

Description=Thermal Daemon Service


[Service]

Type=dbus

SuccessExitStatus=1

BusName=org.freedesktop.thermald

ExecStart=/usr/sbin/thermald --no-daemon --dbus-enable


[Install]

WantedBy=multi-user.target

Alias=dbus-org.freedesktop.thermald.service

It doesn't require the thermal zone/RAPL information to be ready and that's the reason why it doesn't take any effect for thermal throttling.

If I rebuild the kernel with the following CONFIGs from m to y, then the thermal throttling work as expected.

CONFIG_INT340X_THERMAL=m

CONFIG_ACPI_THERMAL_REL=m


CONFIG_INTEL_RAPL_CORE=m

CONFIG_INTEL_RAPL=m

Any suggestion about what I can do except rebuilding the kernel?

Run
https://github.com/intel/dptfxtract/dptfxtract
once. This creates some configuration. Also make sure that you are running atleast thermald version 1.9.1.

Then collect the same log. What is the make and model of this laptop?

Thanks,
Srinivas




Thanks,
Chris

**Comment by mschiu77** [comment]
Hi Srinivas,
    Please refer to the attached journal log. The symptom persists but now it at least uses the generated /etc/thermald/thermal-conf.xml.auto. The log shows a little difference as follows.

```
Jan 20 14:29:56 endless thermald[429]: [WARN]Using generated /etc/thermald/thermal-conf.xml.auto
Jan 20 14:29:56 endless thermald[429]: [WARN]sysfs open failed
Jan 20 14:29:56 endless thermald[429]: [ERR]Thermal DTS or hwmon: No Zones present Need to configure manually
Jan 20 14:29:56 endless thermald[429]: [ERR]XML zone: invalid sensor type TCPU
Jan 20 14:29:56 endless thermald[429]: [WARN]Zone update failed: unable to bind
Jan 20 14:29:56 endless thermald[429]: [ERR]XML zone: invalid sensor type SEN4
Jan 20 14:29:56 endless thermald[429]: [WARN]Zone update failed: unable to bind
```

Seems that it still fail to find the corresponding sysfs entries for the sensors in the configuration file?

[Connex_L1430_journal.log.log](https://github.com/intel/thermal_daemon/files/4084744/Connex_L1430_journal.log.log)

**Comment by spandruvada** [comment]
Hi,


# systemctl stop thermald


Attach output of
#  grep -r . /sys/class/thermal/*


Then run
# thermald ---no-daemon --loglevel=info


And attach the full output of the above command,

Thanks,
Srinivas

thermal-conf.xml.auto. The log shows a little difference as follows.

Jan 20 14:29:56 endless thermald[429]: [WARN]Using generated /etc/thermald/thermal-conf.xml.auto

Jan 20 14:29:56 endless thermald[429]: [WARN]sysfs open failed

Jan 20 14:29:56 endless thermald[429]: [ERR]Thermal DTS or hwmon: No Zones present Need to configure manually

Jan 20 14:29:56 endless thermald[429]: [ERR]XML zone: invalid sensor type TCPU

Jan 20 14:29:56 endless thermald[429]: [WARN]Zone update failed: unable to bind

Jan 20 14:29:56 endless thermald[429]: [ERR]XML zone: invalid sensor type SEN4

Jan 20 14:29:56 endless thermald[429]: [WARN]Zone update failed: unable to bind

Seems that it still fail to find the corresponding sysfs entries for the sensors in the configuration file?

Connex_L1430_journal.log.log<https://github.com/intel/thermal_daemon/files/4084744/Connex_L1430_journal.log.log>

**Comment by mschiu77** [comment]
Hi Srinivas,
    The thermald service will run w/o problem after stop and then run again. Because at this time, all entries in /sys/class/thermal are ready just as the attached ` grep -r . /sys/class/thermal/*` log you want me to provide.
[thermal_info.log](https://github.com/intel/thermal_daemon/files/4089347/thermal_info.log)

My major problem is that the `thermald` not working as expected while running as a systemd service at boot up. At that time, the entries in `/sys/class/powercap` and maybe some other required entries are absent yet.

So I modify the `thermald.service` with the following change as you requested.
```
-ExecStart=/usr/sbin/thermald --no-daemon --dbus-enable
-ExecStart=/usr/sbin/thermald --no-daemon --dbus-enable --loglevel=info
``` 
And then reboot to do journal log again and attached in the following
[Connex_L1430_thermal_info.log](https://github.com/intel/thermal_daemon/files/4089405/Connex_L1430_thermal_info.log)

The thermald get `[WARN]sysfs open failed` at 13:59:45 and have the kernel message `proc_thermal 0000:00:00.1: Creating sysfs group for PROC_THERMAL_PCI`. I don't know whether if it's related.

**Comment by spandruvada** [comment]
On Mon, 2020-01-20 at 22:08 -0800, Ming Shuo Chiu wrote:

Hi Srinivas,
The thermald service will run w/o problem after stop and then run again. Because at this time, all entries in /sys/class/thermal are ready just as the attached grep -r . /sys/class/thermal/* log you want me to provide.
thermal_info.log<https://github.com/intel/thermal_daemon/files/4089347/thermal_info.log>

You have to do after startup as the acpi zone will cause you to shutdown
#echo disabled > /sys/class/thermal/thermal_zone0/mode

You can do this as part of ExecStart in the systemd unit file.

My major problem is that the thermald not working as expected while running as a systemd service at boot up. At that time, the entries in /sys/class/powercap and maybe some other required entries are absent yet.

So I modify the thermald.service with the following change as you requested.

-ExecStart=/usr/sbin/thermald --no-daemon --dbus-enable

-ExecStart=/usr/sbin/thermald --no-daemon --dbus-enable --loglevel=info

And then reboot to do journal log again and attached in the following
Connex_L1430_thermal_info.log<https://github.com/intel/thermal_daemon/files/4089405/Connex_L1430_thermal_info.log>

Basically the sysfs is not created for thermald to use.

Try adding something like this in the systemd service file. This may be /lib/systemd/system/thermald.service

in the [Unit] section

[Unit]
Description=Thermal Daemon Service
After=syslog.target network.target

...
..


Thanks,
Srinivas

The thermald get [WARN]sysfs open failed at 13:59:45 and have the kernel message proc_thermal 0000:00:00.1: Creating sysfs group for PROC_THERMAL_PCI. I don't know whether if it's related.

**Comment by mschiu77** [comment]
I added the `After` in [Unit] block as follows. And also tried to replace the `After` with `Requires`.
```
Unit]
Description=Thermal Daemon Service
After=syslog.target network.target
```

Seems that the dependencies stays the same and doesn't change. According to output of `systemctl list-dependencies`, it still depends on dbus.socket. From the `systemd-analyze plot` output, it still runs  before network.target done. I also tried lots of different .target combination but no luck.

**Comment by spandruvada** [comment]
On Tue, 2020-01-21 at 23:01 -0800, Ming Shuo Chiu wrote:

I added the After in [Unit] block as follows. And also tried to replace the After with Requires.

Unit]

Description=Thermal Daemon Service

After=syslog.target network.target

Seems that the dependencies stays the same and doesn't change. According to output of systemctl list-dependencies, it still depends on dbus.socket. From the systemd-analyze plot output, it still runs before network.target done. I also tried lots of different .target combination but no luck.

Not seen this race before. Can you try to add sleep in the ExecStart before calling thermald?

**Comment by spandruvada** [comment]
Please try a change, where I added a sleep if the thermal sysfs is not ready. This is on development_2_0 branch.

https://github.com/intel/thermal_daemon/tree/development_2_0

So after clone checkout development_2_0 branch,
If it fixes, I can merge for the next release.

**Comment by jprvita** [comment]
Hello @spandruvada, thanks for your reply!

At Endless we are trying to support a generic setup that works out-of-the-box on different machines, without user intervention. This is the context of @mschiu77's message above. We are seeing this problem on a Connex L1430 SwiftBook, a Scope SN116PYA and another white-label engineering sample from a third manufacturer.

It seems to me that thermald could do a little better and dynamically detect these drivers being loaded after thermald is already running. Would that be a feasible approach, or am I missing something here? If so, is this something on the upstream roadmap?

While we could work around the problem by building the drivers into the kernel, that is not ideal on a generic kernel configuration targeting multiple machines.

**Comment by spandruvada** [comment]
We could detect dynamically but this is tricky if the thermal action is already in progress. So not trivial,

We never had this issue with race. Try this implementation, hope this will address for time being.

**Comment by mschiu77** [comment]
[journal_new_thermald.log](https://github.com/intel/thermal_daemon/files/4133040/journal_new_thermald.log)
Here's the journal log running thermald built from https://github.com/intel/thermal_daemon/tree/development_2_0. I did see the message `Waiting for thermal sysfs to be ready` which suppose to wait 2 seconds, but the `/dev/acpi_thermal_rel` still absent and the `TRT/ART read failed` after that. And I can still run stress test to trigger the ` thermal thermal_zone0: critical temperature reached (100 C), shutting down`.

**Comment by spandruvada** [comment]
the thermal relationship error will not be solved. You need to run dptfxtract tool once.
But I don't see TCPU and SEN4 loaded. I hope you are running on the same system as before (/sys/class/thermal/thermalzone*/type had SEN4 and TCPU). Try to increase trimeout and check.

Also attach /sys/class/powercap/intel_rapl/intel_rapl:0/*. Looks like information is missing or your are older kernel. Use the latest 5.5 kernel which should have ICL support.

**Comment by mschiu77** [comment]
Sorry. I did test the thermald w/ thermal-conf.xml.* and w/o thermal-conf.xml.*. I forgot to attach the journal log running with the thermal config generated by dptfxtract. Here it is. Although it behaves almost the same as w/o thermal config. The both cases are running under the kernel 5.3.
[kernel5.3_thermal.log](https://github.com/intel/thermal_daemon/files/4149049/kernel5.3_thermal.log)

I'll build the kernel 5.5 and test again

**Comment by dsd** [comment]
All the platforms that @mschiu77 is testing here are Gemini Lake and Apollo Lake.
Would the Linux 5.5 ICL support be expected to make any difference here?

**Comment by spandruvada** [comment]
On Mon, 2020-02-03 at 17:25 -0800, Daniel Drake wrote:

All the platforms that @mschiu77<https://github.com/mschiu77> is testing here are Gemini Lake and Apollo Lake.
Would the Linux 5.5 ICL support be expected to make any difference here?

No.

**Comment by mschiu77** [comment]
[5.5_thermal.log](https://github.com/intel/thermal_daemon/files/4160661/5.5_thermal.log)

Here's the journal log running on latest kernel 5.5. Per the log, it still suffers the same problem. Please suggest what else information I can help collect for you.

**Comment by flixr** [comment]
I'm having the same Problem with the race condition at startup... sometimes thermald hangs because only some devices were there, sometimes it completes startup but does nothing because at that time the RAPL sysfs was not yet there...

Is there anything that we could check to decide when to start thermald?
Ideally some sort of systemd target (which we could create).

**Comment by real-or-random** [comment]
I have a similar issue. 

When starting at boot:
```
Sep 09 16:23:27 systemd[1]: Starting Thermal Daemon Service...
Sep 09 16:23:28 thermald[579]: NO RAPL sysfs present
Sep 09 16:23:28 thermald[579]: 32 CPUID levels; family:model:stepping 0x6:9a:3 (6:154:3)
Sep 09 16:23:28 systemd[1]: Started Thermal Daemon Service.
Sep 09 16:23:28 thermald[579]: NO RAPL sysfs present
Sep 09 16:23:28 thermald[579]: 32 CPUID levels; family:model:stepping 0x6:9a:3 (6:154:3)
Sep 09 16:23:28 thermald[579]: Thermal DTS: No coretemp sysfs found
Sep 09 16:23:28 thermald[579]: sensor id 5 : No temp sysfs for reading raw temp
Sep 09 16:23:28 thermald[579]: sensor id 5 : No temp sysfs for reading raw temp
Sep 09 16:23:28 thermald[579]: sensor id 5 : No temp sysfs for reading raw temp
Sep 09 16:23:28 thermald[579]: Config file /etc/thermald/thermal-conf.xml does not exist
Sep 09 16:23:28 thermald[579]: Config file /etc/thermald/thermal-conf.xml does not exist
Sep 09 16:23:28 thermald[579]: Thermal DTS or hwmon: No Zones present Need to configure manually
Sep 09 16:23:28 thermald[579]: Config file /etc/thermald/thermal-conf.xml does not exist
Sep 09 16:23:28 thermald[579]: Polling mode is enabled: 4
```

when restarting later:
```
Sep 09 16:36:28 thermald[9930]: 32 CPUID levels; family:model:stepping 0x6:9a:3 (6:154:3)
Sep 09 16:36:29 thermald[9930]: 32 CPUID levels; family:model:stepping 0x6:9a:3 (6:154:3)
Sep 09 16:36:29 thermald[9930]: sensor id 27 : No temp sysfs for reading raw temp
Sep 09 16:36:29 thermald[9930]: sensor id 27 : No temp sysfs for reading raw temp
Sep 09 16:36:29 thermald[9930]: sensor id 27 : No temp sysfs for reading raw temp
Sep 09 16:36:29 thermald[9930]: Unsupported condition 58 (UKNKNOWN)
Sep 09 16:36:29 thermald[9930]: Unsupported conditions are present
Sep 09 16:36:29 thermald[9930]: Polling mode is enabled: 4
 ```

I'll try probably try add a sleep command...

**Comment by ivanmrva** [comment]
Same behavior on my machine as in the previous comment (errors during the boot, restarting later seems to be OK). Using thermald v.2.5.1.

**Comment by spandruvada** [comment]
Try the latest version and reopen if the issue still exists.
